pkg-manager/plugin-commands-installation/src/installDeps.ts

```diff
@@ -290,7 +290,7 @@ when running add/update with the --workspace option')
   }
 
   const updatedManifest = await install(manifest, installOpts)
-  if (opts.update === true && opts.save !== false) {
+  if ((opts.update === true || (opts.update !== false && opts.resolutionMode === 'highest')) && opts.save !== false) {
```

Comment:
Make it more intuitive.

```suggestion
  if ((opts.update === true || (opts.update === undefined && opts.resolutionMode === 'highest')) && opts.save !== false) {
```